### PR TITLE
fix: only highest eligible coupon applied

### PR DESCRIPTION
**Pull request by sandratridz**
In POS and E-commerce of ERPNext, it only applies the pricing rule from the highest eligible coupon code entered by the user. Any other lower-value coupon codes entered have no effect.

For example, if a user enters the 20% off coupon code for orders over the value of 1000 when placing an order worth 2000, that 20% off coupon is ignored and no discount is applied. Only if the user specifically enters the code for the 30% off coupon eligible for orders over the value of 1500 will a discount be applied.

##  Steps to reproduce the coupon bug

1. Create two coupons in the system:

    - Coupon code "20OFF" that gives 20% off orders over value of 1000.

    - Coupon code "30OFF" that gives 30% off orders over value of 1500.

2. Add an items to the cart/invoice over the threshold for both coupons (ex: 2,000)  

3. Apply the lower value coupon code on the checkout page (enter "20OFF" in the coupon code field)

4. The subtotal does not change/no discount is applied.

5. Enter the higher value coupon code instead (enter "30OFF") 

6. The appropriate 30% discount based on the order subtotal is now applied.

7. Confirm that when only the lower value coupon code is applied, no discount is given.

8. Confirm that a discount is only applied when the higher value eligible coupon code is used.

### CODE REVIEW

**Review by deepeshgarg007 — APPROVED** [review]

File: erpnext/accounts/doctype/pricing_rule/utils.py [hunk 1]
```diff
@@ -581,6 +581,9 @@ def apply_pricing_rule_on_transaction(doc):
 			if d.price_or_product_discount == "Price":
 				if d.apply_discount_on:
 					doc.set("apply_discount_on", d.apply_discount_on)
+
```
Comment (ruchamahabal) [inline erpnext/accounts/doctype/pricing_rule/utils.py]:
```suggestion

```

File: erpnext/accounts/doctype/pricing_rule/utils.py [hunk 2]
```diff
@@ -611,6 +619,12 @@ def apply_pricing_rule_on_transaction(doc):
 							doc.set(field, 0)
 
 				doc.calculate_taxes_and_totals()
+
+				# Break out of the main loop if the condition is met
+				if condition_met:
+					break
+
+
```
Comment (ruchamahabal) [inline erpnext/accounts/doctype/pricing_rule/utils.py]:
```suggestion

```

**Comment by ankush** [comment]
@mergifyio update

**Comment by ankush** [comment]
@sandratridz can you merge latest develop branch. And also please raise PRs from personal accounts in future. 

https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#maintainer-edits

**Pull request merged**